chore(blog): asyncapi 2024 mentorship program metrics (#4333)
asyncapi/website: markdown/blog/2024-mentorship-metrics.md

---
title: "2024 Mentorship Program Metrics"
date: 2025-08-20T06:00:00+01:00
type: Community
tags:
  - Mentorship 
  - Communication
cover: /img/posts/mentorship-assets/metric-bg.webp
authors:
  - name: Azeez Elegbede
    photo: /img/avatars/ace.webp
    link: https://www.linkedin.com/in/aelegbede/
    byline: AsyncAPI Preacher
excerpt: "AsyncAPI Mentorship Program: A Deep Dive into the Stats"
---

> AsyncAPI mentorship program is an online global opensource program that provides a structured pathway for bridging the gap between the AsyncAPI contributors and it's projects by pairing them with experienced mentors and maintainers over a defined period of time. 

Reviewer: I got a bit confused with the narrative:
> for bridging the gap between the AsyncAPI contributors and it's projects

Did I understand correctly that there is a gap between those who contribute to AsyncAPI and the projects AsyncAPI has as an Initiative? What kind of gap it is? Why do we need to bridge it? I would add a bit of context here so that readers better understand the purpose of the program

Reviewer: @AceTheCreator let's clarify a bit this one

Author: yea, i thought i left a comment here 😄

The gap I was highlighting is the disparity in the quality of contributions made by regular contributors compared to maintainers.

Thanks to your input, I saw that the narrative wasn't working, so I've rewritten the whole passage.

> AsyncAPI mentorship program is an online global opensource program that provides a structured pathway for bridging the gap between the AsyncAPI contributors and it's projects by pairing them with experienced mentors and maintainers over a defined period of time. 

Reviewer: Duplication of the previous paragraph

```suggestion
```

<Figure
  src="/img/posts/mentorship-assets/overview.webp"
  caption="Overview of 2024 Mentorship Program metrics"
  className="text-center"
/>

## Contributors

- 65.04% of selected contributors had little to no experience contributing to open source.
- 80% of contributors were active communicators throughout the program. 
- 77.78% of selected contributors followed best practices while contributing. 

## Projects

- 37.5% of accepted project proposals were aimed at project maintainership.

> - The program achieved a project completion rate of 90%.

Reviewer: ```suggestion
- 90% - a project completion rate.
```

- 85.7% of contributors agreed that mentors provided adequate guidance and support to achieve their project goals.

## Mentors

<Figure
  src="/img/posts/mentorship-assets/pie.webp"
  caption="Mentor composition and evaluation outcomes"
  className="text-center"
/>

The 2024 program welcomed eight highly esteemed and experienced mentors, 80% of whom are active maintainers.

- Two-thirds of accepted mentors are returning mentors from previous editions of the program.

- 85.7% of mentors passed the mentorship evaluations. 

> - 77.78% of mentors provided mentees with an exceptional mentorship experience.

Reviewer: ```suggestion
- Two-thirds of accepted mentors are returning mentors from previous editions of the program.
- 85.7% of mentors passed the mentorship evaluations. 
- 77.78% of mentors provided mentees with an exceptional mentorship experience.
```

## Program

<Figure
  src="/img/posts/mentorship-assets/line-graph.webp"
  caption="Reported learning impact across the program timeline"
  className="text-center"
/>

90% of contributors reported that the program made a meaningful contribution to their learning, helping them gain valuable skills, knowledge, and hands-on experience that supported their growth throughout the program.

### Feedback

> Thank you for giving me to opportunity, and that my background does not define me contributing to OSS.

> The program is already doing well. I believe if the suggestions made by Ace during one of our community meetings to add certifications are included then the program will be even better.

> 

Reviewer: ```suggestion
```

> The program is just amazing for contributors like me who are starting their journey.

> First off, I'll say a very big Thank you!

> I've grown so much in this program, and I owe it

> Please don't take initiatives that are out of your bandwith, like to get some feedback of my work I had to personally dm 5 to 6 people and except derberg and ace sometimes the rest don't give a damn.

> 

Reviewer: ```suggestion
```

> The program goal is clear. But one thing that lukasz always say that we have to focus more on maintainership. Including projects that require maintainership will be very helpful to the community. I agree to this point, it will be giving AsyncAPI community a lot of dedicated contributors.

> The program is very impactful especially to early-career techies like me so I could grow in open source

> So For one, I suggest a better structure that kicks things off immediately the mentees are announced. That way, we still have that excitement we had getting into the program reserved all the way to the end (Or at least the first evaluation)

> nah, everything is perfect.

### Conclusion

A big thank you to everyone who participated in this program—from the kickoff discussions to the wrap-up event. Y'all are the special ones. And to those spreading the word about the AsyncAPI Mentorship Program, you all belong on the cover of Essence.

> Finally, a special shout-out to the mentors for making this edition of the program a success. We'll keep you updated on future entries for the AsyncAPI Maintainership Program this year—stay tuned. 

Reviewer: ```suggestion
Finally, a special shout-out to the mentors for making this edition of the program a success. We'll keep you updated on future entries for the AsyncAPI Maintainership Program this year, so stay tuned. 
```

> Meanwhile, we’ve shared the data from both the first and second evaluation phases for mentors and mentees. If the highlights above don’t already speak for themselves, feel free to dive deeper into the numbers:

Reviewer: ```suggestion
Meanwhile, we've shared the data from both the first and second evaluation phases for mentors and mentees. If the highlights above don’t seem enough, feel free to dive deeper into the numbers:
```

## Projects

- 37.5% of accepted project proposals were aimed at project maintainership.
- 90% - a project completion rate.
- 85.7% of contributors agreed that mentors provided adequate guidance and support to achieve their project goals.

## Mentors

<Figure
  src="/img/posts/mentorship-assets/pie.webp"
  caption="Mentor composition and evaluation outcomes"
  className="text-center"
/>

The 2024 program welcomed eight highly esteemed and experienced mentors, 80% of whom are active maintainers.

- Two-thirds of accepted mentors are returning mentors from previous editions of the program.
- 85.7% of mentors passed the mentorship evaluations. 
- 77.78% of mentors provided mentees with an exceptional mentorship experience.

## Program

<Figure
  src="/img/posts/mentorship-assets/line-graph.webp"
  caption="Reported learning impact across the program timeline"
  className="text-center"
/>

90% of contributors reported that the program made a meaningful contribution to their learning, helping them gain valuable skills, knowledge, and hands-on experience that supported their growth throughout the program.

### Feedback

> > Thank you for giving me to opportunity, and that my background does not define me contributing to OSS.

Reviewer 2: ```suggestion
> Thank you for giving me the opportunity, and my background does not define me, contributing to OSS.
```

> The program is already doing well. I believe if the suggestions made by Ace during one of our community meetings to add certifications are included then the program will be even better.

> The program is just amazing for contributors like me who are starting their journey.

> > First off, I'll say a very big Thank you!

Reviewer 3: ```suggestion
> First off, I'll say a very big thank you!
```

> I've grown so much in this program, and I owe it

> > Please don't take initiatives that are out of your bandwith, like to get some feedback of my work I had to personally dm 5 to 6 people and except derberg and ace sometimes the rest don't give a damn.

Reviewer 2: ```suggestion
> Please don't take initiatives that are out of your bandwidth, like to get some feedback on my work, I had to personally dm 5 to 6 people and except for derberg and ace, sometimes the rest don't give a damn.
```

> > The program goal is clear. But one thing that lukasz always say that we have to focus more on maintainership. Including projects that require maintainership will be very helpful to the community. I agree to this point, it will be giving AsyncAPI community a lot of dedicated contributors.

Reviewer 3: ```suggestion
> The program goal is clear. But one thing that Lukasz always say that we have to focus more on maintainership. Including projects that require maintainership will be very helpful to the community. I agree to this point, it will be giving AsyncAPI community a lot of dedicated contributors.
```

> The program is very impactful especially to early-career techies like me so I could grow in open source

> > So For one, I suggest a better structure that kicks things off immediately the mentees are announced. That way, we still have that excitement we had getting into the program reserved all the way to the end (Or at least the first evaluation)

Reviewer 2: ```suggestion
> So, for one, I suggest a better structure that kicks things off immediately after the mentees are announced. That way, we still have that excitement we had getting into the program reserved all the way to the end (Or at least the first evaluation)
```

> nah, everything is perfect.

### Conclusion

> A big thank you to everyone who participated in this program—from the kickoff discussions to the wrap-up event. Y'all are the special ones. And to those spreading the word about the AsyncAPI Mentorship Program, you all belong on the cover of Essence.

Reviewer 2: ```suggestion
A big thank you to everyone who participated in this program, from the kickoff discussions to the wrap-up event. Y'all are the special ones. And to those spreading the word about the AsyncAPI Mentorship Program, you all belong on the cover of Essence.
```
